plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/procedure/TestIcebergAddFilesProcedure.java

```diff
@@ -0,0 +1,537 @@
+    {
+        String hiveTableName = "test_add_files_" + randomNameSuffix();
+        String icebergTableName = "test_add_files_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE hive.tpch." + hiveTableName + " AS SELECT CAST(NULL AS int) x", 1);
+        assertUpdate("CREATE TABLE iceberg.tpch." + icebergTableName + "(x int NOT NULL)");
+
+        String path = (String) computeScalar("SELECT \"$path\" FROM hive.tpch." + hiveTableName);
+        assertQueryFails(
+                "CALL iceberg.system.add_files(target_schema_name=>'tpch', target_table_name=>'" + icebergTableName + "', location=>'" + path + "', format=>'ORC')",
+                ".*NULL value not allowed for NOT NULL column: x");
+        assertQueryFails(
+                "CALL iceberg.system.add_files('tpch', '" + icebergTableName + "', 'tpch', '" + hiveTableName + "')",
+                ".*NULL value not allowed for NOT NULL column: x");
+
+        assertQueryReturnsEmptyResult("SELECT * FROM iceberg.tpch." + icebergTableName);
+
+        assertUpdate("DROP TABLE hive.tpch." + hiveTableName);
+        assertUpdate("DROP TABLE iceberg.tpch." + icebergTableName);
+    }
+
+    @Test
+    void testAddFilesDifferentFileFormat()
+    {
+        String hiveTableName = "test_add_files_" + randomNameSuffix();
+        String icebergTableName = "test_add_files_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE iceberg.tpch." + icebergTableName + " WITH (format = 'PARQUET') AS SELECT 1 x", 1);
+        assertUpdate("CREATE TABLE hive.tpch." + hiveTableName + " WITH (format = 'ORC') AS SELECT 2 x", 1);
+
+        assertUpdate("CALL iceberg.system.add_files('tpch', '" + icebergTableName + "', 'tpch', '" + hiveTableName + "')");
+
+        assertQuery("SELECT * FROM " + icebergTableName, "VALUES 1, 2");
+
+        assertUpdate("DROP TABLE hive.tpch." + hiveTableName);
+        assertUpdate("DROP TABLE iceberg.tpch." + icebergTableName);
+    }
+
+    @Test
+    void testAddFilesAcrossSchema()
+    {
+        String hiveSchemaName = "test_schema" + randomNameSuffix();
+        String hiveTableName = "test_add_files_" + randomNameSuffix();
+        String icebergTableName = "test_add_files_" + randomNameSuffix();
+
+        assertUpdate("CREATE SCHEMA hive." + hiveSchemaName);
+
+        assertUpdate("CREATE TABLE iceberg.tpch." + icebergTableName + " WITH (format = 'PARQUET') AS SELECT 1 x", 1);
+        assertUpdate("CREATE TABLE hive." + hiveSchemaName + "." + hiveTableName + " WITH (format = 'ORC') AS SELECT 2 x", 1);
+
+        assertUpdate("CALL iceberg.system.add_files('tpch', '" + icebergTableName + "', '" + hiveSchemaName + "', '" + hiveTableName + "')");
+
+        assertQuery("SELECT * FROM iceberg.tpch." + icebergTableName, "VALUES 1, 2");
+
+        assertUpdate("DROP TABLE iceberg.tpch." + icebergTableName);
+        assertUpdate("DROP SCHEMA hive." + hiveSchemaName + " CASCADE ");
+    }
+
+    @Test
+    void testAddFilesDifferentDataColumnDefinitions()
```

Comment:
*suggestion*: this only checks different column name, do we need a different column type check? Never mind, I see `testAddFilesTypeMismatch` that checks exactly this, so maybe move it closer to this one or merge the two?